public/app/features/dashboard-scene/serialization/sceneVariablesSetToVariables.ts

```diff
@@ -275,14 +276,26 @@ export function sceneVariablesSetToSchemaV2Variables(
       }
       const query = variable.state.query;
       let dataQuery: DataQueryKind | string;
+      const datasource = getElementDatasource(set, variable, 'variable', undefined, dsReferencesMapping);
+
       if (typeof query !== 'string') {
         dataQuery = {
-          kind: variable.state.datasource?.type ?? getDataQueryKind(query),
+          kind: 'DataQuery',
+          version: defaultDataQueryKind().version,
+          group: datasource?.type ?? getDataQueryKind(query),
+          datasource: {
+            name: datasource?.uid,
+          },
```

Comment:
guess in this scenario we don't need the `datasource` property altogether.

Reply:
Dominik! I don't understand. Why do you mean?